documentation/modules/appendix_crds.adoc

```diff
@@ -1896,6 +1899,37 @@ Used in: xref:type-BrokerCapacity-{context}[`BrokerCapacity`]
 |Broker capacity for outbound network throughput in bytes per second. Use an integer value with standard Kubernetes byte units (K, M, G) or their bibyte (power of two) equivalents (Ki, Mi, Gi) per second. For example, 10000KiB/s.
 |====
 
+[id='type-HashLoginServiceApiUsers-{context}']
+= `HashLoginServiceApiUsers` schema reference
+
+Used in: xref:type-CruiseControlSpec-{context}[`CruiseControlSpec`]
+
+
+[cols="2,2,3a",options="header"]
+|====
+|Property |Property type |Description
+|type
+|string
+|Type of the Cruise Control API Users configuration. Supported values are: `hashLoginService`.
+|valueFrom
+|xref:type-PasswordSource-{context}[`PasswordSource`]
+|Secret from which the custom Cruise Control API authentication credentials should be read. 
+|====
+
+[id='type-PasswordSource-{context}']
+= `PasswordSource` schema reference
+
+Used in: xref:type-HashLoginServiceApiUsers-{context}[`HashLoginServiceApiUsers`], xref:type-Password-{context}[`Password`]
+
+
+[cols="2,2,3a",options="header"]
+|====
+|Property |Property type |Description
+|secretKeyRef
+|https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#secretkeyselector-v1-core[SecretKeySelector]
+|Selects a key of a Secret in the resource's namespace.
```

Comment:
Could we change this in PasswordSource.java to read: "Selects a key of a secret in the resource's namespace."

Reply:
I thought the policy was to uppercase k8s primitives like k8s "Secret"  objects

Reply:
We use either secret/secrets, pods etc for general reference or backticks and case (`Secret`) for direct reference. I think the general would be fine here